nxdrive/osi/darwin/darwin.py

```diff
@@ -198,28 +202,67 @@ def send_sync_status(self, state: NuxeoDocumentInfo, path: str) -> None:
                 return
 
             name = "{}.syncStatus".format(BUNDLE_IDENTIFIER)
-            status = "unsynced"
-
-            readonly = (os.stat(path).st_mode & (stat.S_IWUSR | stat.S_IWGRP)) == 0
-            if readonly:
-                status = "locked"
-            elif state:
-                if state.error_count > 0:
-                    status = "error"
-                elif state.pair_state == "conflicted":
-                    status = "conflicted"
-                elif state.local_state == "synchronized":
-                    status = "synced"
-                elif state.pair_state == "unsynchronized":
-                    status = "unsynced"
-                elif state.processor != 0:
-                    status = "syncing"
-
-            log.trace("Sending status %r for file %r to FinderSync", status, path)
-            self._send_notification(name, {"status": status, "path": path})
+            status = self._formatted_status(state, path)
+
+            self._send_notification(name, {"statuses": [status]})
+            log.trace(f"Sending status to FinderSync for {path!r}: {status}")
+        except:
+            log.exception("Error while trying to send status to FinderSync")
+
+    @if_frozen
+    def send_content_sync_status(self, states: List[DocPair], path: str) -> None:
+        """
+        Send the sync status of the content of a folder to the FinderSync.
+
+        :param states: current local states of the children of the folder
+        :param path: full path of the folder
+        """
+        try:
+            path = force_decode(path)
+            if not os.path.exists(path):
+                return
+
+            name = "{}.syncStatus".format(BUNDLE_IDENTIFIER)
+
+            # We send the statuses of the children by batch in case
+            # the notification center doesn't allow notifications
+            # with a heavy payload.
+            # 50 seems like a good balance between payload size
+            # and number of notifications.
+            batch_size = 50
```

Comment:
Perhaps allowing to override this number is a good idea? It can be an option in `Options.py`.